Cargo.lock

```diff
@@ -2951,9 +2966,19 @@ dependencies = [
 
 [[package]]
 name = "lightning-persister"
-version = "0.0.103"
+version = "0.0.104"
+dependencies = [
+ "bitcoin",
+ "libc",
+ "lightning",
+ "winapi",
+]
+
+[[package]]
+name = "lightning-persister"
+version = "0.0.104"
```

Comment:
Maybe we can avoid using the external `lightning-persister` crate?

Reply:
> Isn't it strange that there are two lightning-persister with version = "0.0.104"? :D

I imported the `lightning-persister` crate code to mm2 to be able to implement external backup for lightning channels. I kept the version as "0.0.104" to be able to track new changes to the crate and apply it to the imported code in the future with every new `rust-lightning` release.

As for The external crate, it's a dependency for `lightning-background-processor` https://github.com/KomodoPlatform/atomicDEX-API/blob/05c66a10428b315874e8ce829736b3512bb87f4c/Cargo.lock#L2940

> Maybe we can avoid using the external lightning-persister crate?

To avoid this, I would need to import the `lightning-background-processor` crate to mm2 as well. I guess I will import it in the future anyways for Wasm support along other crates, but I wanted to delay this until working on Wasm support to avoid tracking changes to these other crates as well.
I see that the `lightning-background-processor` doesn't have a lot of code, so I will import it now as it won't be difficult to track changes to it.

Reply:
Done

Reply:
@artemii235 the `lightning-background-processor` tests add files and folders and then remove them at the end of every test by using the drop trait https://github.com/KomodoPlatform/atomicDEX-API/blob/0da2ccb1b279ff4db079d90542172702a27af71f/mm2src/coins/lightning_background_processor/src/lib.rs#L413-L421 Is this behavior acceptable for the CI? Just want to be sure so I can also add the `lightning-persister` tests from `rust-lightning`